htlcswitch/link.go

```diff
@@ -1850,6 +1850,20 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		}
 
 	case *lnwire.UpdateFailHTLC:
+		// Verify that the failure reason is at least 256 bytes plus
+		// overhead.
+		const minimumFailReasonLength = lnwire.FailureMessageLength +
+			2 + 2 + 32
+
+		if len(msg.Reason) < minimumFailReasonLength {
```

Comment:
If we received such a short reason before this change, what would happen?

We must be sure this is not creating any incompatibilities we didn't have before.

Reply:
We'd just relay it back to the sender, and risk triggering the `len == 260` rule. I think we can make this change safely as the other implementations acknowledged that they only use 256 bytes failure messages?

Reply:
Assume the channels `Alice<->Bob<->Carol`, and that Alice updates to the version of LND that has this added check (Bob has not updated). Could Carol get Alice to close her channel by maliciously sending a short error that Bob would happily relay?

Reply:
Hmmm, I see. Good point. Maybe we should just pad the failure here, to at least avoid triggering the length-based malformed failure detection?

Reply:
Yes, the reason for padding is that we'd at least pass back valid length messages. That opens up the possibility to be stricter about this later down the line.

Reply:
What does padding mean here? Only padding (and not adding a mac) to trigger this error
- (https://github.com/lightningnetwork/lightning-onion/blob/b62f49fcadfc013d81c639e196a7f09b428a4dc1/crypto.go#L317-L319)

Reply:
Updated commit to return a compliant-length array of random bytes.

Reply:
could add a test for this to ensure that if we a node receives < 292, they send backwards a fail message with 292 bytes?

Reply:
Added